### ISSUE TITLE
File Realm Authentication Error 

### ISSUE THREAD

**Issue by CrispyBaconElastic** [elastic/elasticsearch#93332]
### Elasticsearch Version

8.6.1

### Java Version

_bundled_

### OS Version

MacOS Ventura

### Problem Description

In the attached observability stack,  when the observability stack is loaded using docker-compose up, the users and roles are added using file mounts to the elasticsearch configuration directory in the elasticsearch pod.  These users/roles work for authentication however are not visible in the users application from Kibana.  The authentication for these users should require the password "testing", but "changeme" works for all of the users in the file.  
```bash
admin:$2a$10$xiY0ZzOKmDDN1p3if4t4muUBwh2.bFHADoMRAWQgSClm4ZJ4132Y.
apm_server_user:$2a$10$iTy29qZaCSVn4FXlIjertuO8YfYVLCbvoUAJ3idaXfLRclg9GXdGG
apm_user_ro:$2a$10$hQfy2o2u33SapUClsx8NCuRMpQyHP9b2l4t3QqrBA.5xXN2S.nT4u
beats_user:$2a$10$LRpKi4/Q3Qo4oIbiu26rH.FNIL4aOH4aj2Kwi58FkMo1z9FgJONn2
filebeat_user:$2a$10$sFxIEX8tKyOYgsbJLbUhTup76ssvSD3L4T0H6Raaxg4ewuNr.lUFC
heartbeat_user:$2a$10$nKUGDr/V5ClfliglJhfy8.oEkjrDtklGQfhd9r9NoFqQeoNxr7uUK
kibana_system_user:$2a$10$nN6sRtQl2KX9Gn8kV/.NpOLSk6Jwn8TehEDnZ7aaAgzyl/dy5PYzW
metricbeat_user:$2a$10$5PyTd121U2ZXnFk9NyqxPuLxdptKbB8nK5egt6M5/4xrKUkk.GReG
``` 
[local_lab.zip](https://github.com/elastic/elasticsearch/files/10530703/local_lab.zip)

### Steps to Reproduce

1.  Extract the attached zip file.
2.  "docker-compose up" from the observability_stack directory
3. After the cluster has started attempt to authenticate "admin:changeme" to the kibana UI.
4. Navigate to users in stack management
5. Verify the available users/roles

**Comment by ywangd** [comment]
> The authentication for these users should require the password "testing", but "changeme" works for all of the users in the file.

But the `admin` entry in your users file
```
admin:$2a$10$xiY0ZzOKmDDN1p3if4t4muUBwh2.bFHADoMRAWQgSClm4ZJ4132Y.
```
does have `changeme` as its password. So it is working correctly. 

>  These users/roles work for authentication however are not visible in the users application from Kibana. 

File based users and roles are not retrievable using GetUsers/GetRoles API nor visible in Kibana's users page. This is explicilty called out in the [documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/file-realm.html#file-realm) (see the "important" note). 

In summary, both features are working as intended. As such, I hope you don't mind me closing this issue.
